src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -236,4 +250,46 @@ private static boolean checkIsAllLinesAreWhitespace(String[] lines, int lineFrom
         }
         return result;
     }
+
+    /**
+     * Checks for empty LITERAL_CASE.
+     *
+     * @param ast a {@code DetailAST} value
+     * @return true iff case is empty
+     */
+    private boolean isCaseEmpty(DetailAST ast) {
+        final DetailAST slistToken = ast.getNextSibling();
+        boolean returnValue = false;
+        if (slistToken != null
+                && slistToken.getType() == TokenTypes.SLIST) {
+            DetailAST leftCurly;
+            if (option == BlockOption.STMT) {
+                for (leftCurly = slistToken.getFirstChild();
```

Comment:
Can we use TokenUtils#findFirstTokenByPredicate here?
Just think what predicate do you need for this.